Babystep_Multiplicator_Fix

AnHardt opened the pull request:
Analog to https://github.com/MarlinFirmware/MarlinDev/pull/250 by
@RicardoGA

Restore the Babystep Multiplicator functionality and now it's available
for X/Y/Z

Wackerbarth commented:
This is not a Bug Fix. It is an enhancement.
This breaks the spirit of version numbering. Someone who has a working 1.1-RC2 system which enables the option will not be able to continue to use their existing configuration file.

AnHardt commented:
@Wackerbarth 
When `BABYSTEP_Z_MULTIPLICATOR` is not evaluated at all is absolutely useless. Having it in the config or not makes no difference. 
How about adding `#define BABYSTEP_MULTIPLICATOR 1` instead of 2.  Then the current behavior will not change.
On the way from 1.0.2 to 1.1.0 we had and will have a lot of changes in the configuration. Until we have a final 1.1.0 _I_ don't see a problem with further changes.

Wackerbarth commented:
No, I still say this is an enhancement. If you keep adding features, then we will never release 1.1.0 because there will always be "one more problem that needs fixing". We really need to focus on releasing the features that do work.

This addition can properly be considered for 1.1.1 since, as you point out, it can be added with a default that requires no change to the configuration and is therefore backward compatible.

AnHardt commented:
Hmmm.
Recently I have not seen anyone working on bugs but me. And I (like we all) decide what i'm working on. At least for me it looks, as if you are playing with new features for the IDE in 1.1.1. Even on bug reports about that you do not react, until i try to fix them. So it makes very little difference if i'm working on this or that. Here at the RC , at least occasionally i have some testers.

You are right, in the sense of, every squeezed bug is an enchantment.

AnHardt commented:
@RicardoGA
Sorry, this is your FIX. I try to push it here to motivate you for more of your good work.

Rikrdoga commented:
@AnHardt that's ok, im currently working with marlin dev branch in my printer, with this fix/enhancement and your fix Reimplement BABYSTEP_INVERT_Z #267 . and the print counter feature that im planning to add another counter to keep the record how many prints  start vs prints  finish. 

PS: if you need help to make some test or something let me know

AnHardt committed:
Babystep_Multiplicator_Fix

Analog to https://github.com/MarlinFirmware/MarlinDev/pull/250 by
@RicardoGA

Restore the Babystep Multiplicator functionality and now it's available
for X/Y/Z

AnHardt commented:
Changed the `#define BABYSTEP_MULTIPLICATOR` to 1 everywhere to not alter the former behavior.
Rebased.

AnHardt commented:
Aloso fixing https://github.com/MarlinFirmware/Marlin/issues/2742 if set negative.
